x/group/server/invariants_test.go

```diff
@@ -339,3 +341,263 @@ func TestGroupTotalWeightInvariant(t *testing.T) {
 		require.Equal(t, spec.expErr, broken)
 	}
 }
+
+func TestProposalTallyInvariant(t *testing.T) {
+	curCtx, cdc, key := getCtxCodecKey(t)
+
+	// Group Account Table
+	groupAccountTableBuilder := orm.NewPrimaryKeyTableBuilder(GroupAccountTablePrefix, key, &group.GroupAccountInfo{}, orm.Max255DynamicLengthIndexKeyCodec{}, cdc)
+	groupAccountTable := groupAccountTableBuilder.Build()
+
+	// Group Member Table
+	groupMemberTableBuilder := orm.NewPrimaryKeyTableBuilder(GroupMemberTablePrefix, key, &group.GroupMember{}, orm.Max255DynamicLengthIndexKeyCodec{}, cdc)
+	groupMemberByGroupIndex := orm.NewUInt64Index(groupMemberTableBuilder, GroupMemberByGroupIndexPrefix, func(val interface{}) ([]uint64, error) {
+		group := val.(*group.GroupMember).GroupId
+		return []uint64{group}, nil
+	})
+	groupMemberTable := groupMemberTableBuilder.Build()
+
+	// Proposal Table
+	proposalTableBuilder := orm.NewAutoUInt64TableBuilder(ProposalTablePrefix, ProposalTableSeqPrefix, key, &group.Proposal{}, cdc)
+	proposalTable := proposalTableBuilder.Build()
+
+	// Vote Table
+	voteTableBuilder := orm.NewPrimaryKeyTableBuilder(VoteTablePrefix, key, &group.Vote{}, orm.Max255DynamicLengthIndexKeyCodec{}, cdc)
+	voteTable := voteTableBuilder.Build()
+
+	_, _, adminAddr := testdata.KeyTestPubAddr()
+	_, _, addr1 := testdata.KeyTestPubAddr()
+	_, _, addr2 := testdata.KeyTestPubAddr()
+
+	curBlockTime, err := gogotypes.TimestampProto(curCtx.BlockTime())
+	require.NoError(t, err)
+
+	specs := map[string]struct {
+		groupAccReq []*group.GroupAccountInfo
+		policy      group.DecisionPolicy
+		membersReq  []*group.GroupMember
+		proposalReq []*group.Proposal
+		voteReq     []*group.Vote
+		expErr      bool
+	}{
+		"invariant not broken": {
+			groupAccReq: []*group.GroupAccountInfo{
+				{
+					Address: addr1.String(),
+					GroupId: 1,
+					Admin:   adminAddr.String(),
+					Version: 1,
+				},
+			},
+			policy: group.NewThresholdDecisionPolicy(
+				"1",
+				gogotypes.Duration{Seconds: 1},
+			),
+			membersReq: []*group.GroupMember{
+				{
+					GroupId: 1,
+					Member: &group.Member{
+						Address: addr1.String(),
+						Weight:  "2",
+					},
+				},
+				{
+					GroupId: 1,
+					Member: &group.Member{
+						Address: addr2.String(),
+						Weight:  "1",
+					},
+				},
+			},
+			proposalReq: []*group.Proposal{
+				{
+					ProposalId:          1,
+					Address:             addr1.String(),
+					Proposers:           []string{addr1.String()},
+					SubmittedAt:         *curBlockTime,
+					GroupVersion:        1,
+					GroupAccountVersion: 1,
+					Status:              group.ProposalStatusSubmitted,
+					Result:              group.ProposalResultUnfinalized,
+					VoteState:           group.Tally{YesCount: "2", NoCount: "1", AbstainCount: "0", VetoCount: "0"},
```

Comment:
```suggestion
					Member: &group.Member{
						Address: addr1.String(),
						Weight:  "3",
					},
				},
				{
					GroupId: 1,
					Member: &group.Member{
						Address: addr2.String(),
						Weight:  "1",
					},
				},
			},
			proposalReq: []*group.Proposal{
				{
					ProposalId:          1,
					Address:             addr1.String(),
					Proposers:           []string{addr1.String()},
					SubmittedAt:         *curBlockTime,
					GroupVersion:        1,
					GroupAccountVersion: 1,
					Status:              group.ProposalStatusSubmitted,
					Result:              group.ProposalResultUnfinalized,
					VoteState:           group.Tally{YesCount: "3", NoCount: "1", AbstainCount: "0", VetoCount: "0"},
```

if you change the test case to this, the invariant shouldn't be broken but with your current implementation, the test fails

Reply:
okay